File: backend/CrossStitchViewer/Mappers/UserMapper.cs
using CrossStitchViewer.Models;
using Data.Records;

namespace CrossStitchViewer.Mappers;

public static class UserMapper
{
    public static UserModel Map(UserRecord user) => new()
    {
        Reference = user.Reference,
        CreatedAt = user.CreatedAt,
        Email = user.Email,
        Username = user.Username
    };
}

File: frontend/src/api/mappers/User.mapper.ts
import dayjs from 'dayjs';

import { type IUser } from '@/models/User.model';
import { type IApiUser } from '@/api/api-models/ApiUser.type';

export const userMapper = {

    map(user: IApiUser): IUser {
        return {
            reference: user.reference,
            createdAt: dayjs(user.createdAt),
            email: user.email,
            lastLoginAt: user.lastLoginAt === null ? null : dayjs(user.lastLoginAt),
        };
    },

};